docs/elasticsearch-spec.asciidoc

```diff
@@ -359,64 +359,59 @@ The init container inherits the image of the main container image if one is not
 
-* `maxSurge` specifies the number of Pods that can be added to the cluster, on top of the desired number of nodes in the specification during cluster updates
-* `maxUnavailable` specifies the number of Pods that can be made unavailable during cluster updates
-
-The default of `maxSurge: 1; maxUnavailable: 0` spins up an additional Elasticsearch node during cluster updates.
-It is possible to speed up cluster topology changes by increasing `maxSurge`. For example, setting `maxSurge: 3` would allow 3 new nodes to be created while the original 3 migrate data in parallel.
-The cluster would then temporarily have 6 nodes.
-
-Setting `maxSurge` to 0 and `maxUnavailable` to a positive value only allows a maximum number of Pods to exist on the Kubernetes cluster.
-For example, `maxSurge: 0; maxUnavailable: 1` would perform the 3 nodes upgrade this way:
-
-1. Migrate data away from the first 16GB node
-2. Once data is migrated, remove the 16GB node: the cluster temporarily has 2 nodes
-3. Add a new 32GB node: the cluster grows to 3 nodes
-4. Follow the same steps for the 2 other 16GB nodes
 
-Even if a `changeBudget` is specified, ECK makes sure that some invariants are maintained while a mutation is in progress. In the cluster, there must be at least:
+==== Caveats
+* Setting both `maxSurge` and `maxUnavailable` to `0` will prevent the operator from performing any update that needs to bring down or create a new pod.
+* Due to the safety measures employed by the operator, a particular changeBudget might be insufficient to perform the requested update. For example, removing last data node from one nodeSet and adding data node to a different nodeSet is not possible with `maxSurge` set to 0. In this case operator can't remove the node (because it can't be migrated since no other data nodes are available) and can't add the node (because `maxSurge` is set to 0).
+* Currently, the operator has limited understanding of the cluster topology and, during update, might not pick the optimal order of pod additions and removals. This may lead to update being stuck and require specifying higher `maxSurge` than theoretically needed.
 
-* One master node alive
-* One data node alive
+In presence of any of the above, operator will emit logs similar to the following:
 
-So under certain circumstances ECK ignores the change budget. For example, a safe migration from a 1-node cluster to another 1-node cluster can only be done by temporarily setting up a 2-nodes cluster.
+```
+Limiting nodes creation to respect maxSurge setting   {(...) "target": 2, "actual": 0}
+```
 
-It is possible to configure the `changeBudget` to optimize the reuse of persistent volumes, instead of migrating data across nodes. This feature is not supported yet, more details to come in the next release.
+```
+Cannot downscale StatefulSet  {(...) "reason": "Not removing node to respect maxUnavailable setting"}
```

Comment:
Not sure we should copy the logs content in the documentation, pretty sure they will change at some point and we won't update the docs accordingly.

Reply:
Wanted to give users something to look for in the logs, but I guess I can be less explicit and convey the same thing. Fixed.